docs/DeveloperGuide.md

```diff
@@ -1187,28 +1196,92 @@ The `MathUtil` class contains generic functions for mathematical operations such
 
 ---
 
-## **Unit Tests**
+## **Testing Types**
+
+Types of tests used:
+* Developer Testing
+* Regression Testing
+* Unit Tests
+* Stubs
+* Integration testing
+* Dogfooding
+
+Our organisation made use of different testing types in order to cover multiple grounds. Types of testings also changed 
+with the progress of the product.
+
+### **Developer Testing**
+
+Our organisation opted to use developer testing right from the start with every increment of the product, instead of 
+waiting for the whole product to be ready before conducting more tests.
+
+This type of testing involved the developers trying out the product with the various commands implemented, to 
+discover possible bugs early, and avoid major reworks in future increments and iterations.
+
+### **Regression Testing**
+
+Our organisation continued with regression testing as increments came in, to ensure that previous features were not 
+broken with changes and modifications. This is due to the interconnectedness of certain features implemented, such as 
+commands and the model. We used a breadth-first iterative model to extensive test the wide scope of the project.
+
+Our organisation did this both with automated unit tests and manual testing, with more emphasis on unit tests, due to 
+the time taken to test extensively manually. 
+
+### **Unit Tests**
 
 Our organisation primarily utilises `JUnit5` for conducting unit tests. `JUnit5` is a popular testing framework in the Java programming language that enables developers to write and run unit tests for their code.
```

Comment:
The tenses should be the same throughout. Please standardise.
Line 1214 - Our organisation **opted** for...
Line 1222 - Our organisation **continued** with...
Line 1226 - Our organisation **did** this ...

Then this line:
Our organisation primarily **utilises**